core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/TypeUtils.kt

```diff
@@ -383,21 +386,35 @@ internal fun <T> getValuesType(values: List<T>, type: KType, infer: Infer): KTyp
     }
 
 /**
- * Returns the value type of the given [values] sequence.
+ * Returns the guessed value type of the given [values] sequence.
+ *
+ * This function analyzes all [values] once and returns the expected column type.
```

Comment:
Did we cover in tests the situation when one column in file or in List has different types?

Reply:
We do have a lot of tests of this specific function: https://github.com/Kotlin/dataframe/blob/nulls-in-framecols/core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/types/UtilTests.kt.
These cases are probably covered by other column creation tests, like in https://github.com/Kotlin/dataframe/blob/nulls-in-framecols/core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/constructors.kt